File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.js
function solution(n, t, m, p) {
    let fullString = "";
    let num = 0;
    
    while(fullString.length < t * m){
        fullString += (num++).toString(n).toUpperCase()
    }
    
    let result = "";
    for(let i = 0 ; i < t ; i++){
        result += fullString[i * m + (p-1)];
    }
    return result;
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def convert(num, base):
    digits = "0123456789ABCDEF"
    
    if num == 0:
        return "0"
    
    result = ""
    while num > 0:
        result = digits[num % base] + result
        num //= base
    return result

def solution(n, t, m, p):
    result = ""
    full_string = ""
    number = 0
    
    while len(full_string) < t * m:
        full_string += convert(number, n)
        number += 1

    for i in range(t):
        result += full_string[i * m + (p - 1)]

    return result